.github/workflows/test.yml

```diff
@@ -8,16 +8,25 @@ env:
 jobs:
   build:
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        GO_VERSION:
+          - "1.17.8"
+          - "1.20.4"
+          - "1.20.5"
```

Comment:
The go.mod file says "go 1.12". I feel like we should either test on go1.12, or update that to say "go 1.17". I believe there were significant language changes between those versions that we may accidentally violate.